cogs/fun.py

```diff
@@ -422,48 +424,81 @@ async def bill(self, ctx):
 
 			await ctx.send(url + tag['href'])
 
+
 	@commands.group(name='xkcd', invoke_without_command=True)
 	@commands.bot_has_permissions(embed_links=True)
-	async def xkcd(self, ctx, *, id: int = None):
-		'''Get a random or specified xkcd comic.'''
+	async def xkcd(self, ctx, *, search : typing.Union[int, str] = None):
+		'''Gets an xkcd comic.'''
 		async with ctx.typing():
-			if id is None:
+			if search is None:
 				await self.random(ctx)
-				return
-			url = f'https://xkcd.com/{id}/info.0.json'
-			async with ctx.http.get(url) as resp:
-				if resp.status != 200:
-					raise commands.CommandError('Request failed.')
-				comic_json = await resp.json()
-			e = await self.make_xkcd_embed(comic_json)
+			elif isinstance(search, int):
+				await self.num(ctx, search)
+			else:
+				await self.relevant(ctx, search=search)
+
+	@commands.cooldown(rate=3, per=10.0, type=commands.BucketType.user)
+	@xkcd.command(aliases=['id','#'])
+	async def num(self, ctx, id):
+		'''Shows a specific xkcd comic given a number'''
+		async with ctx.typing():
+			e = await self.get_xkcd_comic(ctx, id)
 			await ctx.send(embed=e)
 
 
-	@xkcd.command(aliases=('today','recent','now'))
+	@commands.cooldown(rate=3, per=10.0, type=commands.BucketType.user)
+	@xkcd.command(aliases=['today','recent','now'])
 	async def latest(self, ctx):
+		'''Returns the latest xkcd comic from xkcd.com'''
 		async with ctx.typing():
 			url = 'https://xkcd.com/info.0.json'
 			comic_json = await self.get_xkcd_json(ctx,url)
 			e = await self.make_xkcd_embed(comic_json)
 			await ctx.send(embed=e)
 	
-	@xkcd.command()
+	@commands.cooldown(rate=3, per=10.0, type=commands.BucketType.user)
+	@xkcd.command(aliases=['rand'])
 	async def random(self, ctx):
+		'''Returns a random xkcd comic'''
 		async with ctx.typing():
 			url = 'https://c.xkcd.com/random/comic/'
 			async with ctx.http.get(url) as resp:
-				if resp.status == 404:
-					raise commands.CommandError('Comic does not exist.')
 				if resp.status != 200:
 					raise commands.CommandError('Request failed.')
-				url = str(resp.url)
-				num = url.lstrip('https://xkcd.com/').rstrip('/')
+				num = str(resp.url).lstrip('https://xkcd.com/').rstrip('/')
 			url = 'https://xkcd.com/{}/info.0.json'.format(num)
 			comic_json = await self.get_xkcd_json(ctx, url)
 			e = await self.make_xkcd_embed(comic_json)
 			await ctx.send(embed=e)
-			return
 	
+	@commands.cooldown(rate=1, per=7.0, type=commands.BucketType.user)
+	@xkcd.command(aliases=['search'])
+	async def relevant(self, ctx, *, search : str):
+		'''Grabs a relevant xkcd from `relevantxkcd.appspot.com`.'''
```

Comment:
```suggestion
		'''Get a relevant xkcd comic from `relevantxkcd.appspot.com`.'''
```

Reply:
Or without the backticks if the URL then becomes a link. Would be appropriate imo.

Reply:
> 
> 
> Or without the backticks if the URL then becomes a link. Would be appropriate imo.

To make it a link it would have to be in markdown form, so why not.

```suggestion
		'''Grabs a relevant xkcd from [`relevantxkcd.appspot.com`](https://relevantxkcd.appspot.com).'''
```